worker/runtime/process.go

```diff
@@ -48,6 +51,10 @@ func (p *Process) Wait() (int, error) {
 
 	p.process.IO().Wait()
 
+	if p.stdin != nil {
+		p.stdin.Close()
```

Comment:
I'll have to do some more testing on this, but this whole `func (*stdinWrapper) Close()` thing might not even be needed. The whole point of it is to notify the `stdinWrapper` to cleanup after itself by erroring out of read.

But looking at https://github.com/containerd/containerd/blob/1c05317a47686692143837d07d8df4702a8bd2c7/cio/io_unix.go#L65-L71, if a containerd process exits, the fifo will get closed, `io.CopyBuffer()`should error, which would cause the goroutine to exit, and (hopefully) go's gc will come along and get rid of the `stdinWrapper` anyways.

Reply:
No wait `io.CopyBuffer()` blocks on `Read`, which means that we have to `Close` it somehow. I wonder if it's possible for containerd to miss the process exiting, in which case this could cause a memory leak.

> Either way, maybe we could Close stdin prior to calling Wait() in case that behaviour changes to match the docs? 

My thought was that since this is supposed to be nop, let's wait until we know for sure the process has exited before cleaning up. That way there's no way of this causing any weird concurrency problems by potentially sending a SIGHUP to an exited process but before the `stdout` and `stderr` are drained.